dali/base/dautils.cpp

```diff
@@ -1998,7 +2013,21 @@ class CTimedCache
         if (!item)
             return 0;
         CriticalBlock block(sect);
-        item->due = msTick()+PAGE_CACHE_TIMEOUT;
+        if ((maxPageCacheItems > 0) && (maxPageCacheItems == items.length()))
+        {
+            unsigned indexToRemove = items.length() - 1;
+            unsigned minExpires = 0xFFFFFFFF;
+            ForEachItemInRev(i,items)
+            {
+                CTimedCacheItem &item = items.item(i);
+                if (item.due < minExpires)
+                {
+                    minExpires = item.due;
+                    indexToRemove = i;
+                }
+            }
+            items.remove(indexToRemove, true);
```

Comment:
I am concerned this is costly, i.e. every add incurs a complete scan of the cache..

A better approach would be to ensure that the tail items are the most recent, i.e. the head is oldest (or vice versa).
Then this pruning only involves removing 1 item without a scan.

And in fact, since add() is always called, whether a new item or an old, the newest will always be at tail already, i.e. oldest always at head.
So you can simply remove the head without a scan afaics.

Reply:
Also, not new, but check() is prob. not coded correctly for similar reasons.
It is scanning in reverse, despite the fact that newer entries are at tail and it's looking for older entries.
It could scan forward and when it finds an entry that's not expired it could stop, since all ahead of it won't have expired either.

If you change this, please verify the above.

Reply:
I just verified that the order is from old to new (the oldest is at the head).